### DISCUSSION TITLE
Feature Request: Import GitHub Stars

### DISCUSSION THREAD

**Discussion by Scarvy** [Textualize/frogmouth#80] [Ideas]
Hi,

Would a feature to import .md files from a user's starred repos be in the scope of this project?

I was thinking of either a new tab in the navigation panel called "GitHub Stars" or directly importing the .md files into the bookmark bar as I show below:

[![asciicast](https://asciinema.org/a/aEL7Q3ARpUt7z2oQfXBMFPhhC.svg)](https://asciinema.org/a/aEL7Q3ARpUt7z2oQfXBMFPhhC)

**Comment by davep** [comment]
I suspect this would require access to the GitHub API, which would require logins and tokens and the like, and also would start to tie the application to GitHub somewhat (you'll notice we've tried to ensure it works with a few git forges); my immediate reaction is that this feels like it's not within the scope of a Markdown viewer.

However, the format and the location of the bookmark file is stable, so a third party "import bookmarks" tool would be a fun addition for someone to write.

**Comment by Scarvy** [comment]
Thank you for the feedback and totally understand maintaining the broader compatibility of the viewer.

The idea of creating a separate "import bookmarks" tool is intriguing. I'm open to exploring this further.
